### PR TITLE
Delete obsolete (older than 180 days) Analysis entries.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/analyzer/backend.dart [hunk 1]
```diff
@@ -166,4 +166,27 @@ class AnalysisBackend {
 
     return true;
   }
+
+  Future deleteObsoleteAnalysisEntries() async {
+    final DateTime threshold =
+        new DateTime.now().toUtc().subtract(const Duration(days: 180));
```
Comment (kevmoo) [inline app/lib/analyzer/backend.dart]: Good thing you used `toUtc` here

File: app/bin/service/analyzer.dart [hunk 2]
```diff
@@ -111,3 +113,18 @@ Future _initFlutterSdk() async {
 void _registerServices() {
   registerAnalysisBackend(new AnalysisBackend(db.dbService));
 }
+
+final Random _random = new Random.secure();
```
Comment (kevmoo) [inline app/bin/service/analyzer.dart]: Good thing this is secure

File: app/lib/analyzer/backend.dart (hunk 1 above)
Comment (isoos) [inline app/lib/analyzer/backend.dart]: @mkustermann has a strict policy on it :)

File: app/bin/service/analyzer.dart (hunk 2 above)
Comment (isoos) [inline app/bin/service/analyzer.dart]: I once worked with an audit guy for a short project. He loved to flag the non-secure random things as 'high-risk' security issues, regardless of the context, and it doesn't harm too much. So why not? :)

Comment (kevmoo) [inline app/bin/service/analyzer.dart]: Yup. Just being silly...

File: app/lib/analyzer/backend.dart (hunk 1 above)
Comment (kevmoo) [inline app/lib/analyzer/backend.dart]: 😄

**Pull request merged**